src/config.ts

```diff
@@ -0,0 +1,128 @@
+import { relative } from 'pathe'
+import { addTemplate, createResolver, useNuxt } from '@nuxt/kit'
+import loadConfig from 'tailwindcss/loadConfig.js'
+import resolveConfig from 'tailwindcss/resolveConfig.js'
+import logger from './logger'
+import type { ModuleHooks, ModuleOptions, TWConfig } from './types'
+import configMerger from './runtime/merger.mjs'
+
+type PartialTWConfig = Partial<TWConfig>
+
+const serializeConfig = <T extends PartialTWConfig>(config: T) =>
+JSON.stringify(
+  Array.isArray(config.plugins) && config.plugins.length > 0 ? configMerger({ plugins: (defaultPlugins: TWConfig['plugins']) => defaultPlugins?.filter((p) => p && typeof p !== 'function') }, config) : config,
+  (_, v) => typeof v === 'function' ? `() => (${JSON.stringify(v())})` : v).replace(/"(\(\) => \(.*\))"/g, (_, substr) => substr.replace(/\\"/g, '"')
+)
+
+/**
+ * Loads all possible Tailwind CSS configuration for a Nuxt project.
+ *
+ * @param moduleOptions configuration for the module
+ * @param nuxt nuxt app
+ *
+ * @returns template with all configs handled
+ */
+export default async function loadTwConfig(configPaths: string[], contentPaths: string[], inlineConfig: ModuleOptions['config'], nuxt = useNuxt()) {
```

Comment:
we may need to have this run after `app:templatesGenerated` or something. `prepare` would be required; may be a bigger breaking change unless something can be done in Nuxt core.